edb/schema/ordering.py

```diff
@@ -294,6 +296,13 @@ def _trace_op(op, opstack, depgraph, renames, renames_r, strongrefs,
             # happen first.
             deps.add(('delete', op.classname))
 
+            if isinstance(obj, s_func.Function):
+                old_funcs = old_schema.get_functions(
+                    sn.shortname_from_fullname(op.classname),
+                    default=[])
+                for old_func in old_funcs:
+                    deps.add(('delete', old_func.get_name(old_schema)))
+
```

Comment:
I'm not sure I follow.  What if you have two functions:

```
foo(a: float64) -> int64
foo(a: int64) -> int64
```

and you want to alter one of them, e.g.

```ALTER FUNCTION foo(a: int64) ...```

Wouldn't this code mean that `foo(a: float64)` variant gets deleted?

Reply:
No.  This applies only to the commands *already* in the migration. We must sort `DROP FUNCTION foo` before `CREATE FUNCTION foo`, otherwise the `CREATE FUNCTION` will complain that an incompatible signature already exists.

Reply:
ok.